tests/robustness/model/non_deterministic_test.go

```diff
@@ -19,116 +19,18 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/assert"
-
-	"go.etcd.io/etcd/api/v3/mvccpb"
 )
 
 func TestModelNonDeterministic(t *testing.T) {
-	type testOperation struct {
-		req     EtcdRequest
-		resp    EtcdNonDeterministicResponse
-		failure bool
+	nonDeterministicTestScenarios := []nonDeterministicModelTest{}
+	for _, tc := range deterministicModelTestScenarios {
+		nonDeterministicTestScenarios = append(nonDeterministicTestScenarios, toNonDeterministicTest(tc))
 	}
-	tcs := []struct {
-		name       string
-		operations []testOperation
-	}{
-		{
-			name: "First Get can start from non-empty value and non-zero revision",
-			operations: []testOperation{
-				{req: getRequest("key"), resp: getResponse("key", "1", 42, 42)},
-				{req: getRequest("key"), resp: getResponse("key", "1", 42, 42)},
-			},
-		},
-		{
-			name: "First Range can start from non-empty value and non-zero revision",
-			operations: []testOperation{
-				{req: rangeRequest("key", true, 0), resp: rangeResponse([]*mvccpb.KeyValue{{Key: []byte("key"), Value: []byte("1")}}, 1, 42)},
-				{req: rangeRequest("key", true, 0), resp: rangeResponse([]*mvccpb.KeyValue{{Key: []byte("key"), Value: []byte("1")}}, 1, 42)},
-			},
-		},
-		{
-			name: "First Range can start from non-zero revision",
-			operations: []testOperation{
-				{req: rangeRequest("key", true, 0), resp: rangeResponse(nil, 1, 1)},
-				{req: rangeRequest("key", true, 0), resp: rangeResponse(nil, 1, 1)},
-			},
-		},
-		{
-			name: "First Put can start from non-zero revision",
-			operations: []testOperation{
-				{req: putRequest("key", "1"), resp: putResponse(42)},
-			},
-		},
-		{
-			name: "First delete can start from non-zero revision",
-			operations: []testOperation{
-				{req: deleteRequest("key"), resp: deleteResponse(0, 42)},
-			},
-		},
-		{
-			name: "First Txn can start from non-zero revision",
-			operations: []testOperation{
-				{req: compareRevisionAndPutRequest("key", 0, "42"), resp: compareRevisionAndPutResponse(false, 42)},
-			},
-		},
-		{
-			name: "Get response data should match put",
-			operations: []testOperation{
-				{req: putRequest("key1", "11"), resp: putResponse(1)},
-				{req: putRequest("key2", "12"), resp: putResponse(2)},
-				{req: getRequest("key1"), resp: getResponse("key1", "11", 1, 1), failure: true},
```

Comment:
I think it's caused by that `toNonDeterministicTest` changes failure from `true` to `false`.